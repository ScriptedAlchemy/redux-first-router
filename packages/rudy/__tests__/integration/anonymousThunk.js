import createTest from '../../__helpers__/createTest'

createTest(
  'anonymous thunks can be dispatched',
  {
    SECOND: {
      path: '/second',
      beforeEnter: async ({ dispatch }) => {
        await dispatch(async ({ dispatch }) => dispatch({ type: 'REDIRECTED' }))
      },
    },
  },
  async ({ dispatch, getState }) => {
    // test outside of route callbacks:
    const res = await dispatch(({ dispatch }) =>
      dispatch({ type: 'REDIRECTED' }),
    )

    expect(res).toMatchSnapshot()
    expect(getState()).toMatchSnapshot()
  },
)

createTest(
  'anonymous thunks can return actions for automatic dispatch',
  {
    SECOND: {
      path: '/second',
      beforeEnter: ({ dispatch }) => dispatch(() => ({ type: 'REDIRECTED' })),
    },
  },
  async ({ dispatch, getState }) => {
    // test outside of route callbacks:
    const res = await dispatch(() => ({ type: 'REDIRECTED' }))

    expect(res).toMatchSnapshot()
    expect(getState()).toMatchSnapshot()
  },
)
