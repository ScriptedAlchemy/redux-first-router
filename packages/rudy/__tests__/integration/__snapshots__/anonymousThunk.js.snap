// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous thunks can be dispatched "SECOND": 1 - response 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`anonymous thunks can be dispatched "SECOND": 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`anonymous thunks can be dispatched "SECOND": 1 - routes - REDIRECTED - onComplete 1`] = `1`;

exports[`anonymous thunks can be dispatched "SECOND": 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`anonymous thunks can be dispatched "SECOND": 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`anonymous thunks can be dispatched "SECOND": 1 - title 1`] = `"REDIRECTED"`;

exports[`anonymous thunks can be dispatched snipes 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`anonymous thunks can be dispatched snipes 2`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - response 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - routes - REDIRECTED - onComplete 1`] = `1`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`anonymous thunks can return actions for automatic dispatch "SECOND": 1 - title 1`] = `"REDIRECTED"`;

exports[`anonymous thunks can return actions for automatic dispatch snipes 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`anonymous thunks can return actions for automatic dispatch snipes 2`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;
